scheduler/task.go

```diff
@@ -353,6 +354,16 @@ func (t *task) Stop() {
 	defer t.Unlock()
 	if t.state == core.TaskFiring || t.state == core.TaskSpinning {
 		t.state = core.TaskStopping
+		for {
+			_, ok := <-t.isFiring
```

Comment:
Closing the `isFiring` channel happens in `schedulerWorkflow.Start()`, exactly [here](https://github.com/intelsdi-x/snap/pull/1592/files#diff-1d62bbee71bcbb0a2a124a3047dd2aa8R245). 

@rashmigottipati, I have couple concerns about that in existing form:
- maybe do that in `task.fire()` would be more accurate
- based on the `isFiring` channel is not for sending data (data is irrelevant in this case), consider changing its declaration to `chan struct{}`, not bool type

I still have doubts about if this `isFiring` is really necessary - my question is why locking task in [fire()](https://github.com/intelsdi-x/snap/pull/1592/files#diff-77f163e381dd113e90ebd1bd76479b54R538) and in [Stop()](https://github.com/intelsdi-x/snap/pull/1592/files#diff-77f163e381dd113e90ebd1bd76479b54R353) is not enough? @rashmigottipati, could you explain it to me more thoroughly?